### PR TITLE
Prevent next action from covering error

### PR DESCRIPTION
**Pull request by sengsational**
Issue #263
Description:
New method signatures that include a runnable for BaseActivity 

Changes:
Pass a runnable into the `showErrorMessage` method so that the next action is shown *after* the user sees the dialog.

### CODE REVIEW

**Comment by sengsational** [comment]
Daniel, I think maybe the continuous integration thing is broken.  What do you think?

**Comment by kalaspuffar** [comment]
I added the WIP (work in progress) to the title just to see when the pull request is ready for review. Please remove when you are done and add it back if you work on the issue so I know the status.

**Comment by sengsational** [comment]
Daniel, I think that this is ready to go.  I think the continuous integration is falsely reporting a problem.  If you see that there is a "real" problem with this change, can you please help me determine what the CI tool thinks is a problem?  It looks to me as if the CI tool is timing-out, and the research I've done suggest that there's a configuration problem with the CI tool when the CI tool times-out.

**Comment by kalaspuffar** [comment]
I think I got the build system up and running again so a rebase could be good.

**Comment by kalaspuffar** [comment]
Hi @sengsational 

Great to hear that the test on your device didn't fail. So does this mean you think this solution is sufficient and we should keep it as is or did the point I raised in my review add any value?

Just asking because I'm not sure that we are talking about the same thing or that you read the question.

When the review is over I will merge this pull request so I wait for your answer.

Best regards
Daniel

**Comment by sengsational** [comment]
From my perspective, the proposed change does work (prevents the next login dialog from covering up the error dialog).

With respect to :

> "I'm not sure that we are talking about the same thing or that you read the question"

Could you please repeat the question or point me to it?  I don't see a question that needs to be answered.

File: app/src/main/java/org/ea/sqrl/activites/base/LoginBaseActivity.java [hunk 1]
```diff
@@ -172,12 +172,17 @@ public void onTextChanged(CharSequence password, int start, int before, int coun
...
+                            final Runnable nextAction = new Runnable() {
+                                @Override
+                                public void run() {
+                                    showLoginPopup();
+                                }
+                            };
+                            showErrorMessage(R.string.decrypt_identity_fail, nextAction);
                             handler.post(() -> {
                                 txtLoginPassword.setHint(R.string.login_identity_password);
                                 txtLoginPassword.setText("");
```
[the remark this answers is not in the document]
Comment (sengsational) [reply app/src/main/java/org/ea/sqrl/activites/base/LoginBaseActivity.java]: That change having to do with nextAction runner might be beyond my understanding of how this application is coded.   The approach of this pull request can be discarded and I can learn by seeing the alternative way to solve the problem, when that gets implemented.   Certainly this pull request highlights the point where the current code falls down.

[the remark this answers is not in the document]
Comment (kalaspuffar) [reply app/src/main/java/org/ea/sqrl/activites/base/LoginBaseActivity.java]: Hi @sengsational 

I like the approach and we might merge it as is. The reason I asked is that we don't need to reset these fields until we show the popup. 

The handler feature is there as a standard Android feature to handle calls to the main thread. So you create a handler in the main thread and then post GUI operations to that handler to ensure that all the GUI related logic runs on the main thread.

Best regards
Daniel

**Pull request merged**